### PR TITLE
Multi-BUID registrations

### PR DESCRIPTION
**Pull request by Kobzol**
TODO: kill switch semantics

### CODE REVIEW

**Review by jendakol — APPROVED** [review]
For what I (untouched with TS) can say, it's good.

File: functions/src/functions/register-buid.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+const Request = t.type({
+    platform: t.string,
+    platformVersion: t.string,
+    manufacturer: t.string,
+    model: t.string,
+    locale: t.string,
+    pushRegistrationToken: t.string
+});
+
+function parseRequest(data: any): {} {
```
Comment (petr-k) [inline functions/src/functions/register-buid.ts:26]: What about introducing a static `Request` type using (`io-ts`'s `TypeOf`) and annotating the return type of `parseRequest` with it?

File: functions/src/functions/register-buid.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import * as admin from "firebase-admin";
+import {PathReporter} from "io-ts/lib/PathReporter";
+import {MAX_BUIDS_PER_USER, REGION} from "../settings";
+import {CollectionReference} from "@google-cloud/firestore";
+import {randomBytes} from "crypto";
+
+const MAX_BUID_RETRIES = 10;
+const BUID_BYTE_LENGTH = 10;
+
+function getUTCTimestamp(): number {
```
Comment (petr-k) [inline functions/src/functions/register-buid.ts:13]: Minor nitpick: there's technically no such thing as a _UTC timestamp_. A correct term here is _Unix timestamp_.

**Review by petr-k — APPROVED** [review]

File: functions/src/functions/register-buid.ts (hunk 1 above)
Comment (Kobzol) [reply functions/src/functions/register-buid.ts:26]: Wow, this is pretty cool, thanks. I was looking for something like this.

**Pull request merged**